app/services/idv/analytics.rb

```diff
@@ -0,0 +1,40 @@
+module Idv
+  class Analytics
+    attr_reader :base_analytics, :user
+
+    def initialize(base_analytics:, user:)
+      @base_analytics = base_analytics
+      @user = user
+    end
+
+    def idv_review_complete(**kwargs)
+      base_analytics.idv_review_complete(**kwargs, **common_analytics_attributes)
+    end
```

Comment:
This could be implemented with some metaprogramming like [`method_missing`](https://ruby-doc.org/core-2.6.3/BasicObject.html#method-i-method_missing) to avoid having to list out every method, though that would be much more "magical" (in a not-strictly-positive sense) where I opted for the explicit instead, for now anyways.

No strong feeling one way or the other if there's a preference.

Reply:
I think it's better to list out every method!

Reply:
As much a fan I am of the explictness, as I'm starting to work through the list of events, I'm becoming more and more inclined to do _something_ to address the repetition and error-prone nature of listing out each event. This was also a realization I came to after recognizing that the analytics regression specs in `spec/features/idv/analytics_spec.rb` only cover "happy path" and not any of the branching paths.

Contemplating one of either:

1. Allowlisting the name of events which should be decorated and enhance them using `method_missing`
2. Wrapping `track_event` in the decorator and auto-enhance any method prefixed with `'Idv:'`

```rb
class AnalyticsDecorator
  DECORATED_METHODS = [
    :idv_final,
    # ...
  ]

  def respond_to_missing(name)
    DECORATED_METHODS.include?(name)
  end

  def method_missing(name, **kwargs)
    analytics.send(name, **kwargs, **common_analytics_attributes)
  end
end
```


```rb
class AnalyticsDecorator
  def track_event(event, attributes = {})
    attributes.merge!(common_analytics_attributes) if event.start_with?(/idv:/i)
    analytics.track_event(event, attributes)
  end
end
```

Reply:
I revised this in 813e0ebe84f8bbd38fd72e58073b28ff2b05d720 to dynamically define the methods. On the upside, I still think there's some explicitness at least in that the methods are listed out in the constant. I'm not overly attached to this, but it'll at least simplify the initial pass at implementing all the methods, if we still want to choose to break them out again later.